src/reanimated2/screenTransition/swipeSimulator.ts

```diff
@@ -0,0 +1,228 @@
+'use strict';
+import type {
+  PanGestureHandlerEventPayload,
+  ScreenTransitionConfig,
+  RNScreensTurboModuleType,
+} from './commonTypes';
+import { applyStyle } from './styleUpdater';
+
+const RNScreensTurboModule: RNScreensTurboModuleType =
+  global.RNScreensTurboModule || {
+    startTransition: (_stackTag) => {
+      'worklet';
+      console.warn('[Reanimated] RNScreensTurboModule has not been found.');
+      return {
+        topScreenId: -1,
+        belowTopScreenId: -1,
+        canStartTransition: false,
+      };
+    },
+    updateTransition: (_stackTag, _progress) => {
+      'worklet';
+      console.warn('[Reanimated] RNScreensTurboModule has not been found.');
+    },
+    finishTransition: (_stackTag, _isCanceled) => {
+      'worklet';
+      console.warn('[Reanimated] RNScreensTurboModule has not been found.');
+    },
+  };
+
+function computeEasingProgress(
+  startingTimestamp: number,
+  distance: number,
+  velocity: number
+) {
+  'worklet';
+  if (Math.abs(distance) < 1) {
+    return 1;
+  }
+  const elapsedTime = (_getAnimationTimestamp() - startingTimestamp) / 1000;
+  const currentPosition = velocity * elapsedTime;
+  const progress = currentPosition / distance;
+  return progress;
+}
+
+function easing(x: number): number {
+  'worklet';
+  // based on https://easings.net/#easeOutQuart
+  return 1 - Math.pow(1 - x, 5);
+}
+
+function computeProgress(
+  screenTransitionConfig: ScreenTransitionConfig,
+  event: PanGestureHandlerEventPayload,
+  isTransitionCanceled: boolean
+) {
+  'worklet';
+  const screenDimensions = screenTransitionConfig.screenDimensions;
+  const progressX = Math.abs(event.translationX / screenDimensions.width);
+  const progressY = Math.abs(event.translationY / screenDimensions.height);
+  const progress = isTransitionCanceled
+    ? Math.max(progressX, progressY) / 2
+    : Math.max(progressX, progressY);
+  return progress;
+}
+
+function maybeScheduleNextFrame(
+  step: () => void,
+  didScreenReachDestination: boolean,
+  screenTransitionConfig: ScreenTransitionConfig,
+  event: PanGestureHandlerEventPayload,
+  isTransitionCanceled: boolean
+) {
+  'worklet';
+  if (!didScreenReachDestination) {
+    const stackTag = screenTransitionConfig.stackTag;
+    const progress = computeProgress(
+      screenTransitionConfig,
+      event,
+      isTransitionCanceled
+    );
+    RNScreensTurboModule.updateTransition(stackTag, progress);
+    requestAnimationFrame(step);
+  } else if (screenTransitionConfig.onFinishAnimation) {
+    screenTransitionConfig.onFinishAnimation();
+  }
```

Comment:
Another way to express it would be:
```suggestion
  } else {
    screenTransitionConfig.onFinishAnimation?.();
  }
```